app/code/Magento/Downloadable/Observer/UpdateLinkPurchasedObserver.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Downloadable\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class UpdateLinkPurchasedObserver implements ObserverInterface
+{
+    /**
+     * Core store config
+     * @var \Magento\Framework\App\Config\ScopeConfigInterface
+     */
+    protected $_scopeConfig;
+
+    /**
+     * @var \Magento\Downloadable\Model\ResourceModel\Link\Purchased\CollectionFactory
+     */
+    protected $_purchasedFactory;
```

Comment:
Please, avoid using underscores for marking the property/method scope (https://www.php-fig.org/psr/psr-2/#42-properties). I know that in the legacy code we have the same approach but we should avoid it.

Reply:
property/method names updated.